src/settings/plugins/ifcfg-rh/nms-ifcfg-rh-reader.c

```diff
@@ -4046,158 +4046,143 @@ wireless_connection_from_ifcfg (const char *file,
 }
```

Comment:
Maybe. The extreme counter-example would be a rule, that no function can have more than 3 expression-statements or control-flow-statements. Otherwise would the rule require that code must be split into static functions. Clearly, that would be possible, but totally unreadable.

Likewise, I don't agree that a excessively long function is *necessarily* bad style. If there is no re-usable code, and the entire function consists of a series of steps which is performed in a logical simple manner (no complex nested loops, or complex gotos), splitting it into smaller static functions may provide little benefit. Often that applies to [parsing code](https://gitlab.gnome.org/GNOME/NetworkManager-openvpn/blob/d80d163002f936152cd3ef2230a15efa0683af18/properties/import-export.c#L776).

Whether a static function (which is called only once!) is useful, depends on the circumstance and personal preference.

Reply:
> Likewise, I don't agree

Makes it sound like I don't agree with you, although you didn't say that. I meant: "I don't agree with the common wisdom about long functions ...".